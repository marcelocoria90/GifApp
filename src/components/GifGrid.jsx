 
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem } from './GifItem'


export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs( category )

  return (
    <>
      <h3>{ category }</h3>
      { isLoading && (<p>Cargando...</p>) }

      <div className='card-grid'>
        {
          images.map( (img) => (
            <GifItem 
              key={ img.id }
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
