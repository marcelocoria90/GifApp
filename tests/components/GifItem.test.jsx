/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Prueba en componente <GifItem />', () => { 

  const title = 'bart'
  const url = 'https://media3.giphy.com/media/XuDlhFtiWXZEk/giphy.gif?cid=9775a22f2gpmiv31hmp8wx8bfx18y3tl28ddu5whlo9iuqsh&ep=v1_gifs_search&rid=giphy.gif&ct=g'

  test('debe de hacer match con el snapshop', () => { 
    const { container } = render(<GifItem title={ title } url={ url } />)

    expect( container ).toMatchSnapshot()
   })
   
  test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
    render( <GifItem title={ title } url={ url } />)
    const { src, alt } = screen.getByRole('img')
    expect( src ).toBe( url )
    expect( alt ).toBe( alt )
   })

  test('debe mostrar el titulo en el componente', ()=> {
    render( <GifItem title={ title } url={ url } />)
    expect( screen.getByText( title ) ).toBeTruthy()
  })
 })