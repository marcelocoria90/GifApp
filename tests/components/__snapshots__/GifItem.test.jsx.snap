// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en componente <GifItem /> debe de hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="bart"
      src="https://media3.giphy.com/media/XuDlhFtiWXZEk/giphy.gif?cid=9775a22f2gpmiv31hmp8wx8bfx18y3tl28ddu5whlo9iuqsh&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      bart
    </p>
  </div>
</div>
`;
